core/app/c/[communitySlug]/stages/manage/actions.ts

```diff
@@ -154,14 +217,34 @@ export const updateStageName = defineServerAction(async function updateStageName
 });
 
 export const revalidateStages = defineServerAction(async function revalidateStages() {
+	const loginData = await getLoginData();
+	if (!loginData || !loginData.user) {
+		return ApiError.NOT_LOGGED_IN;
+	}
+
 	revalidateTagsForCommunity(["stages", "PubsInStages"]);
 });
 
 export const addAction = defineServerAction(async function addAction(
 	stageId: StagesId,
 	actionName: Action
 ) {
-	// TODO: add authorization check
+	const loginData = await getLoginData();
+	if (!loginData || !loginData.user) {
+		return ApiError.NOT_LOGGED_IN;
+	}
+
+	const { user } = loginData;
+
+	const authorized = await userCan(
+		Capabilities.manageStage,
+		{ type: MembershipType.stage, stageId },
+		user.id
+	);
```

Comment:
not sure if this is the right one for `addAction`...

Reply:
i believe it is! and it's also the right one for the other action-actions below